traffic_ops/traffic_ops_golang/role/roles.go

```diff
@@ -394,24 +395,15 @@ func Update(w http.ResponseWriter, r *http.Request) {
 		api.HandleErr(w, r, tx, http.StatusPreconditionFailed, api.ResourceModifiedError, nil)
 		return
 	}
-	rows, err := tx.Query(updateRoleQuery(), roleV4.Name, roleV4.Description, currentRoleName)
+	err = tx.QueryRow(updateRoleQuery(), roleV4.Name, roleV4.Description, currentRoleName).Scan(&roleV4.LastUpdated)
 	if err != nil {
-		usrErr, sysErr, code := api.ParseDBError(err)
-		api.HandleErr(w, r, tx, code, usrErr, fmt.Errorf("updating role: %w", sysErr))
-		return
-	}
-	defer rows.Close()
-	if !rows.Next() {
-		api.HandleErr(w, r, tx, http.StatusNotFound, errors.New("no such role"), nil)
-		return
-	}
-	var lastUpdated time.Time
-	for rows.Next() {
-		if err := rows.Scan(&lastUpdated); err != nil {
-			api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, fmt.Errorf("scanning lastUpdated from role update: %w", err))
+		if err == sql.ErrNoRows {
```

Comment:
comparisons with errors should use [`errors.Is`](https://pkg.go.dev/errors#Is) rather than `==`.